packages/core/contracts/cross-chain-oracle/OracleSpoke.sol

```diff
@@ -0,0 +1,185 @@
+ * contracts deployed on the child chain to interact with this contract as an Oracle. Moreover, this contract gives
+ * child chain contracts the ability to trigger cross-chain price requests to the mainnet DVM. This Spoke knows how
+ * to communicate with the parent chain via a "ChildMessenger" contract which directly communicates with the
+ * "ParentMessenger" on mainnet.
+ * @dev The intended client of this contract is an OptimisticOracle on sidechain that needs price
+ * resolution secured by the DVM on mainnet.
+ */
+contract OracleSpoke is OracleBase, OracleAncillaryInterface, OracleInterface, Lockable {
+    ChildMessengerInterface public messenger;
+
+    event SetChildMessenger(address indexed childMessenger);
+
+    constructor(address _finderAddress, ChildMessengerInterface _messengerAddress) OracleBase(_finderAddress) {
+        messenger = _messengerAddress;
+        emit SetChildMessenger(address(messenger));
+    }
+
+    // This assumes that the local network has a Registry that resembles the mainnet registry.
+    modifier onlyRegisteredContract() {
+        RegistryInterface registry = RegistryInterface(finder.getImplementationAddress(OracleInterfaces.Registry));
+        require(registry.isContractRegistered(msg.sender), "Caller must be registered");
+        _;
+    }
+
+    modifier onlyMessenger() {
+        require(msg.sender == address(messenger), "Caller must be messenger");
+        _;
+    }
+
+    /**
+     * @notice This is called to bridge a price request to mainnet. This method will enqueue a new price request
+     * or return silently if already requested. Price requests are relayed to mainnet (the "Parent" chain) via
+     * the ChildMessenger contract.
+     * @dev Can be called only by a registered contract that is allowed to make DVM price requests. Will mark this
+     * price request as Requested, and therefore able to receive the price resolution data from mainnet.
+     * @param identifier Identifier of price request.
+     * @param time Timestamp of price request.
+     * @param ancillaryData extra data of price request.
+     */
+    function requestPrice(
+        bytes32 identifier,
+        uint256 time,
+        bytes memory ancillaryData
+    ) public override nonReentrant() onlyRegisteredContract() {
+        bool newPriceRequested = _requestPrice(identifier, time, _stampAncillaryData(ancillaryData, msg.sender));
+        if (newPriceRequested) {
+            messenger.sendMessageToParent(abi.encode(identifier, time, _stampAncillaryData(ancillaryData, msg.sender)));
+        }
+    }
+
+    /**
+     * @notice Overloaded function to provide backwards compatibility for legacy financial contracts that do not use
+     * ancillary data.
+     */
+    function requestPrice(bytes32 identifier, uint256 time) public override nonReentrant() onlyRegisteredContract() {
+        bool newPriceRequested = _requestPrice(identifier, time, "0x0");
```

Comment:
I think the empty ancillary data should be represented as `""` not `"0x0"`. see how this is done in `Voting.sol` here: https://github.com/UMAprotocol/protocol/blob/e41ab0d60d5598a6af71405db94f4d2e8479a004/packages/core/contracts/oracle/implementation/Voting.sol#L293